pkg/controller/install/resolver.go

```diff
@@ -28,7 +28,9 @@ type StrategyResolverInterface interface {
 	InstallerForStrategy(strategyName string, opClient operatorclient.ClientInterface, opLister operatorlister.OperatorLister, owner ownerutil.Owner, annotations map[string]string, previousStrategy Strategy) StrategyInstaller
 }
 
-type StrategyResolver struct{}
+type StrategyResolver struct {
+	ProxyInjectorBuilder DeploymentInitializerFuncBuilder
```

Comment:
It might, depending on the pod configuration. We could also decide a static set of `initializers` some of which could be noop. Today there is only one for env.
Ideally we should have a layer that returns a chain of deployment `initializer` func based on the context. What we have today is a quick fix, we can improve the strategy resolver and deployment `initializer`. I have a task that tracks it.